src/main/java/uk/gov/companieshouse/officerfiling/api/service/OfficerFilingServiceImpl.java

```diff
@@ -50,4 +59,73 @@ public Optional<OfficerFiling> get(String officerFilingId, String transactionId)
         return repository.findById(officerFilingId);
     }
 
+    /**
+     * Merges the contents of an OfficerFiling patch into the original record.
+     * @param original The base record
+     * @param patch A record with updated values
+     * @return The merged record
+     */
+    @Override
+    public OfficerFiling mergeFilings(OfficerFiling original, OfficerFiling patch, Transaction transaction) {
+        logger.debugContext(transaction.getId(), "Patching filings", new LogHelper.Builder(transaction.getId())
+                .withCompanyNumber(transaction.getCompanyNumber()).withFilingId(original.getId())
+                .build());
+        HashMap<String,String> fieldMap = new HashMap<>();
+        OfficerFiling mergedFiling = null;
+        // Get the current values of the original and patch filings, patch values will overwrite
+        // Original values
+        extractFields(original, fieldMap, transaction);
+        extractFields(patch, fieldMap, transaction);
+        // JavaTimeModule handles Instant serialisation
+        var mapper = JsonMapper.builder().addModule(new JavaTimeModule()).build();
+
+        var updatedFiling = mapper.createObjectNode();
+        for(Map.Entry<String,String> entry : fieldMap.entrySet()){
+            String field = entry.getKey();
+            String value = entry.getValue();
+            updatedFiling.put(field, value);
+        }
+        var updatedFilingJson = updatedFiling.toString();
+        try {
+            mergedFiling = mapper.readerFor(OfficerFiling.class).readValue(updatedFilingJson);
+        } catch (JsonProcessingException e) {
+            throw new OfficerFilingServiceException("Failed to patch an officer filing for company "
+                    + transaction.getCompanyNumber(), e);
+        }
+        return mergedFiling;
+    }
+
+    /**
+     * Extracts the fields from an OfficerFiling object and adds them to the given map
+     */
+    private void extractFields(OfficerFiling filing, HashMap<String,String> fieldMap, Transaction transaction){
+        Class<OfficerFiling> yourClass = OfficerFiling.class;
+        for (Method method : yourClass.getMethods()) {
```

Comment:
Have you considered using the MapHelper in the utils package here?
It may reduce the amount of code you have to write.

Reply:
Hi Chris. Good point. I've modified the code to make use of the MapHelper class.